src/table/components/head/MenuList/MenuGroup.tsx

```diff
@@ -27,10 +27,66 @@ export const interceptClickOnMenuItems = (menuGroups: MenuItemGroup[], cache: Su
 type SubMenusOpenStatusCache = Record<string, React.Dispatch<React.SetStateAction<boolean>>>;
 let subMenusOpenStatusCache: SubMenusOpenStatusCache = {};
 
-const MenuGroupItems = ({ id, onClick, itemTitle, icon, enabled, subMenus }: HeadCellMenuItem) => {
+const MenuGroupItems = ({ autoFocus, id, onClick, itemTitle, icon, enabled, subMenus }: HeadCellMenuItem) => {
   const [openMenu, setOpenMenu] = useState(false);
   const anchorRef = useRef<HTMLDivElement | null>(null);
 
+  const moveToNextFocusItem = (currentFocus: Element) => {
+    let nextItem = currentFocus.nextElementSibling as HTMLDivElement;
+    if (!nextItem) {
+      nextItem = currentFocus.parentElement?.children[0] as HTMLDivElement;
+    } else if (nextItem.tagName === 'HR') {
+      nextItem = moveToNextFocusItem(nextItem);
+    }
+    return nextItem;
+  };
+
+  const moveToPreviousFocusItem = (currentFocus: Element) => {
+    let previousItem = currentFocus.previousElementSibling as HTMLDivElement;
+    if (!previousItem) {
+      const menuItemAmount = currentFocus?.parentElement?.children.length;
+      (currentFocus?.parentElement?.children[(menuItemAmount as number) - 1] as HTMLDivElement).focus();
+    } else if (previousItem.tagName === 'HR') {
+      previousItem = moveToPreviousFocusItem(previousItem);
+    }
+    return previousItem;
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLLIElement>) => {
```

Comment:
When they look good to you, I will try to add some unit tests.

Reply:
I think you can extract this logic out of component (+ it's functions) for testing purposes

Reply:
I am using RTL to test it

Reply:
yes but i was thinking about since these functions are pure (not doing any react related things), we can extract them out in order to not pollute component.

Reply:
in fact, we have a a11y util file where suits best for these util functions IMO

Reply:
very true

Reply:
exactly, I plan to move it to utils/handle-key-press.ts